test/e2e/nomad/utils.go

```diff
@@ -0,0 +1,230 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"os"
+	"sync"
+	"time"
+
+	nomad "github.com/hashicorp/nomad/api"
+
+	"github.com/hetznercloud/hcloud-go/v2/hcloud"
+)
+
+const InitialVolumeCapacity = 10737418240 // 10GiB
+
+type Cluster struct {
+	hcloudClient *hcloud.Client
+	nomadClient  *nomad.Client
+
+	volumesCreated map[string]struct{}
+	lock           sync.Mutex
+}
+
+func NewCluster() (*Cluster, error) {
+	token := os.Getenv("HCLOUD_TOKEN")
+	if token == "" {
+		return nil, fmt.Errorf("HCLOUD_TOKEN env variable is not set")
+	}
+
+	hcloudOpts := []hcloud.ClientOption{
+		hcloud.WithToken(token),
+		hcloud.WithApplication("nomad-csi-e2e", "v2.11.0"), // x-releaser-pleaser-version
```

Comment:
I would make the version a dev version `.*-dev`, and we might need to update the release-please files to search this marker.

Reply:
We could also omit the version here. In the context of development, you are usually a few commits ahead of the version specified here.

Reply:
Yeah, might be simpler.